src/contract.rs

```diff
@@ -18,139 +24,315 @@ pub fn instantiate(
     info: MessageInfo,
     msg: InstantiateMsg,
 ) -> Result<Response, ContractError> {
-    let state = State {
-        count: msg.count,
-        owner: info.sender.clone(),
-    };
+    deps.api.debug("WASMDEBUG: ls instantiate");
     set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
-    STATE.save(deps.storage, &state)?;
+
+    ASSETS.save(deps.storage, &msg.assets)?;
+
+    let ls_config = LsConfig {
+        active: true,
+        chain_id: msg.chain_id,
+    };
+    LS_CONFIG.save(deps.storage, &ls_config)?;
```

Comment:
We can check here if the passed LST denom, and the chain-id are compatible here just as an additional check. If fact, can accept either of them as input and derive the other.